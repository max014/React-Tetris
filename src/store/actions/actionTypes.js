export const UPDATE = 'UPDATE';
export const SET_INPUT = 'SET_INPUT';
export const END = 'END';
export const SET_SCORES = 'SET_SCORES';
export const START_GAME = 'START_GAME';