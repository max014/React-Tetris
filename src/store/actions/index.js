export {
	getScores,
	postScore
} from './scoreBoard';