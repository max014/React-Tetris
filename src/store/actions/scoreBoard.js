import axios from '../../axios';
import * as actionTypes from './actionTypes';

export const postScore = (score, name) => {
	return () => axios.post( '/tetris', {name: name, score: score});
};

export const getScores = () => {
	return dispatch => {
		axios.get('/tetris')
            .then(response => dispatch(setScores(response.data)));
	};
};

export const setScores = scores => {
	return {
		type: actionTypes.SET_SCORES,
		scores: scores.sort((a, b) => b.score-a.score)
	};
};

export const deleteScore = id => {
	return () => axios.delete( '/tetris/' + id);
};