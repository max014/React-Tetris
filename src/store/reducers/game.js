import * as actionTypes from '../actions/actionTypes';
import { updateGame } from '../../logic/updateGame';
import newGame from '../../logic/newGame';

const initialState = {
	...newGame(),
	scores: [],
	modalUp: false
}

const game = (state = initialState, action) => {
	switch(action.type){
		case actionTypes.START_GAME:
			return {
				...state,
				...newGame(),
				lost: false,
				modalUp: false
			}
		case actionTypes.UPDATE:
			const newState = updateGame(state);
			return {
				...state,
				pieces: newState.pieces,
				board: newState.board,
				input: null,
				score: newState.score,
				lines: newState.lines,
				level: newState.level,
				lost: newState.lost
			};
		case actionTypes.SET_INPUT:
			return {
				...state,
				input: action.input
			};
		case actionTypes.END:
			return {
				...state,
				modalUp: true
			};
		case actionTypes.SET_SCORES:
			const scores = action.scores;
			return {
				...state,
				scores: scores
			}
		default:
			return state;
	}
};

export default game;