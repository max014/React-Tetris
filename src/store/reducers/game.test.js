import game from './game';
import * as actionTypes from '../actions/actionTypes';

describe('game reducer', () => {
    const initialState = game(undefined, {type: actionTypes.START_GAME});
    describe('START_GAME', () => {
        it('starts a new game', () => {
            expect(initialState).toMatchSnapshot();
        });
    });

    describe('UPDATE', () => {
        const nextFrame = game(initialState, {type: actionTypes.UPDATE});
        it('adds the first piece', () => {
            expect(nextFrame.pieces).toHaveLength(1);
        });
        it('sets input to `null`', () => {
            expect(nextFrame.input).toBe(null);
        });
    });
    
    describe('SET_INPUT', () => {
        it('sets the keyboard input', () => {
            expect(game(initialState, {type: actionTypes.SET_INPUT, input: 39}).input).toEqual(39);
        });
    });
    
    describe('END', () => {
        it('sets `modalUp` to true', () => {
            expect(game(initialState, {type: actionTypes.END}).modalUp).toBe(true);
        });
    });

    describe('SET_SCORES', () => {
        it('sets `scores`', () => {
            const scores = [
                {name: '1', score: 1},
                {name: '2', score: 2},
                {name: '3', score: 3},
            ];
            expect(game(initialState, {type: actionTypes.SET_SCORES, scores: scores}).scores).toEqual(scores);
        });
    });
});