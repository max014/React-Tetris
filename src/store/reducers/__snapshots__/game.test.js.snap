// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`game reducer START_GAME starts a new game 1`] = `
Object {
  "board": Array [
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
  ],
  "height": 768,
  "input": null,
  "level": 0,
  "lines": 0,
  "lost": false,
  "modalUp": false,
  "pieces": Array [],
  "score": 0,
  "scores": Array [],
  "step": 38,
}
`;
